apps/condo/domains/meter/schema/MeterReportingPeriod.js

```diff
@@ -0,0 +1,169 @@
+            isRequired: false,
+            knexOptions: { isNotNullable: false },
+            kmigratorOptions: { null: true, on_delete: 'models.CASCADE' },
+        },
+
+        start: {
+            schemaDoc: 'This is the field for client-side business logic and push notifications',
+            type: Integer,
+            defaultValue: 20,
+            isRequired: true,
+            hooks: {
+                validateInput: async ({ context, operation, resolvedData }) => {
+                    if (operation === 'create' || operation === 'update') {
+                        if (resolvedData.start > 31 || resolvedData.start < 1) {
+                            throw new GQLError(ERRORS.INVALID_START, context)
+                        }
+                    }
+                },
+            },
+        },
+
+        finish: {
+            schemaDoc: 'This is the field for client-side business logic and push notifications',
+            type: Integer,
+            defaultValue: 25,
+            isRequired: true,
+            hooks: {
+                validateInput: async ({ context, operation, resolvedData }) => {
+                    if (operation === 'create' || operation === 'update') {
+                        if (resolvedData.finish > 31 || resolvedData.finish < 1) {
+                            throw new GQLError(ERRORS.INVALID_FINISH, context)
+                        } 
+                    }
+                },
+            },
+        },
+
+        b2cApp: {
+            schemaDoc: 'Ref to the B2CApp which used to replace default integration in resident\'s app',
+            type: Relationship,
+            ref: 'B2CApp',
+            isRequired: false,
+            knexOptions: { isNotNullable: false }, // Relationship only!
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+    },
+    kmigratorOptions: {
+        constraints: [
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['organization', 'property'],
+                condition: 'Q(deletedAt__isnull=True)',
+                name: 'meterreportingperiod_unique_organization_property',
+            },
+            {
+                type: 'models.UniqueConstraint',
+                fields: ['organization'],
+                condition:
+                    'Q(deletedAt__isnull=True) &' +
+                    'Q(property__isnull=True)',
+                name: 'meterreportingperiod_unique_organization',
+            },
+        ],
+    },
+    hooks: {
+        resolveInput: async ({ context, operation, existingItem, resolvedData }) => {
+            if (operation === 'create') {
+                if (!resolvedData.organization && resolvedData.property) {
+                    const property = await Property.getOne(context, { id: resolvedData.property })
+
+                    resolvedData.organization = property.organization
+                }
+            }
+
+            return resolvedData
+        },
```

Comment:
property can be null if this period should be for the entire organization. At the same time, the plugin prohibits setting the organization in access when creating an entity. That is, it can only set the organization of the related entity. Therefore, the plugin will not cover all cases.